### DISCUSSION TITLE
How to use sampler2dArray

### DISCUSSION THREAD

**Discussion by peiyangli** [google/filament#6177] [Q&A]
How can I get the length of sampler2dArray in a .mat file?  And is there any "setParameter" to use more than one Textures in c++?

**Comment by bejado** [comment]
You can use [`textureSize`](https://registry.khronos.org/OpenGL-Refpages/es3.1/html/textureSize.xhtml) to get the number of layers in a `sampler2dArray`, though performance may not be optimal. From the C++ side you'd create a single Filament `Texture` with sampler type `SAMPLER_2D_ARRAY` and then pass it to `setParameter`.
